app/routes/task_routes.py

```diff
@@ -1 +1,220 @@
+        return {
+                    "task": {
+                        "id": task.id,
+                        "title": task.title,
+                        "description": task.description,
+                        "is_complete": check_complete(task.completed_at)
+                    }
+                }
+
+    
+
+@tasks_bp.put("/<task_id>")
+def update_task(task_id):
+    task = validate_model(Task, task_id)
+    request_body = request.get_json()
+
+    task.title = request_body['title']
+    task.description = request_body['description']
+
+    if "completed_at" in request_body:
+        task.completed_at = request_body["completed_at"]
+
+    else:
+        task.completed_at = None
+
+    db.session.commit()
+
+    return {
+                "task": {
+                    "id": task.id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": check_complete(task.completed_at)
+                }
+            }
+
+@tasks_bp.delete("/<task_id>")
+def delete_task(task_id):
+    task = validate_model(Task, task_id)
+    db.session.delete(task)
+    db.session.commit()
+
+    return {"details": f"Task {task_id} \"Go on my daily walk 🏞\" successfully deleted"}
+
+# @tasks_bp.patch("/<task_id>/mark_complete")
+# def task_mark_complete(task_id):
+#     task = validate_task(task_id)
+    
+#     task.completed_at = datetime.now()
+
+#     db.session.commit()
+
+#     return {
+#                 "task": {
+#                     "id": task.id,
+#                     "title": task.title,
+#                     "description": task.description,
+#                     "is_complete": check_complete(task.completed_at)
+#                 }
+#             }
+
+@tasks_bp.patch("/<task_id>/mark_incomplete")
+def task_mark_incomplete(task_id):
+    task = validate_model(Task, task_id)
+    
+    task.completed_at = None
+
+    db.session.commit()
+
+    return {
+                "task": {
+                    "id": task.id,
+                    "title": task.title,
+                    "description": task.description,
+                    "is_complete": check_complete(task.completed_at)
+                }
+            }
+
+@tasks_bp.patch("/<task_id>/mark_complete")
+def task_mark_complete_slack(task_id):
+    task = validate_model(Task, task_id)
+
+    task.completed_at = datetime.now()
+    db.session.commit()
+
+    slack_message = {
+        "channel": "task-notifications",
+        "text": f"Someone just completed the task {task.title}"
+
+    }
+
+    # Send the Slack message
+    headers = {"Authorization": f"Bearer {SLACKBOT_TOKEN}"}
+    slack_response = requests.post(
+        "https://slack.com/api/chat.postMessage",
+        json=slack_message,
+        headers=headers
+    )
+
+    # Check Slack response for success
+    if slack_response.status_code != 200 or not slack_response.json().get("ok"):
+        return jsonify({"message": "Failed to send Slack notification"}), 500
```

Comment:
You're the only person I've seen who makes this check and I think it's a great idea!